lib/utils.js

```diff
@@ -317,6 +318,32 @@ function isProhibited(key, prohibitedKeys) {
   return false;
 }
 
+/**
+ * Accept an operator key and return whether it is used for a regular expression query or not
+ * @param {string} operator
+ * @returns {boolean}
+ */
+function isRegExpOperator(operator) {
+  return ['like', 'nlike', 'ilike', 'nilike', 'regexp'].includes(operator);
+}
+
+/**
+ * Accept a RegExp string and make sure that any special characters for RegExp are escaped in case they
+ * create an invalid Regexp
+ * @param {string} str
+ * @returns {string}
+ */
+function escapeRegExp(str) {
+  assert.strictEqual(typeof str, 'string', 'String required for regexp escaping');
+  let success = true;
+  try {
+    new RegExp(str); // try to parse string as regexp
+  } catch (unused) {
+    success = false;
```

Comment:
While I am not very happy about the current implementation, I also don't want to block fixing the source of crashes. 

To make it easier for us to change this behavior in the future and start rejecting requests with unescaped regexp values, I am proposing to print a warning here. For example:

```js
  try {
     new RegExp(str); // try to parse string as regexp
   } catch (unused) {
     console.warn(
       'Auto-escaping invalid RegExp value %j supplied by the caller. ' + 
         'Please note this behavior may change in the future.', 
       str
     );
     success = false;
    }
```

While making changes in this function, can we simplify the implementation as follows?

```js
function escapeRegExp(str) {
  assert.strictEqual(typeof str, 'string', 'String required for regexp escaping');
  try {
    new RegExp(str); // try to parse string as regexp
    return str;
  } catch (unused) {
    console.warn(/*...*/);
    return str.replace(/[\-\[\]\/\{\}\(\)\+\?\.\\\^\$\|]/g, '\\$&');
  }
}
```

Reply:
I can change it right now, because we also want this fix to land.
Would you prefer, just catching the error, and returning it as a SyntaxError: Invalid regular expression to the request?

Reply:
well? 😃

Reply:
I was following the chat here and +1 for @bajtos 's simplified implementation:

```js
function escapeRegExp(str) {
  assert.strictEqual(typeof str, 'string', 'String required for regexp escaping');
  try {
    new RegExp(str); // try to parse string as regexp
    return str;
  } catch (unused) {
    // we can reject with proper message in a follow-up PR
    console.warn(/*...*/);
    return str.replace(/[\-\[\]\/\{\}\(\)\+\?\.\\\^\$\|]/g, '\\$&');
  }
}
```

Reply:
ok @jannyHou! changing it now!